packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2952,9 +2955,36 @@ export default (superClass: Class<Parser>): Class<Parser> =>
           this.resetStartLocationFromNode(expr, typeParameters);
         }
         expr.typeParameters = typeParameters;
+
+        // report error if single type parameter used without trailing comma.
+        if (
+          this.hasPlugin("jsx") &&
```

Comment:
Is it possible to check it only in tsx file? @ozanhonamlioglu 
(#14361 )

Reply:
@nicolo-ribaudo Sounds logical to me, shall I change it with tsx? What is your opinion?

Reply:
TSX is "jsx in TypeScript". This check is only executed when TypeScript is enabled, so it's already only checking for tsx.

Reply:
> TSX is "jsx in TypeScript". This check is only executed when TypeScript is enabled, so it's already only checking for tsx.

In our project, we used the same presets (`@babel/preset-env` + `@babel/preset-react` + `@babel/preset-typescript`) to transform all the files, which caused the `.ts` files to be processed by the jsx plugin, so the generics in the `.ts` files also triggered the `SingleTypeParameterWithoutTrailingComma` error. 

This is a wrong usage, we will improve it. Thank you! @ozanhonamlioglu @nicolo-ribaudo